neo4j/io/_bolt4x0.py

```diff
@@ -489,12 +490,21 @@ def on_records(self, records):
     def on_success(self, metadata):
         """ Called when a SUCCESS message has been received.
         """
-        handler = self.handlers.get("on_success")
-        if callable(handler):
-            handler(metadata)
-        handler = self.handlers.get("on_summary")
-        if callable(handler):
-            handler()
+        if metadata.get("has_more"):
+            if self.connection.state == "streaming_discard_all":
+                handler = self.handlers.get("on_success_has_more_streaming_discard_all")
+                self.connection.state = None
+            else:
+                handler = self.handlers.get("on_success_has_more")
+            if callable(handler):
+                handler(self.connection, **self.handlers)
```

Comment:
Can we get rid of all `is callable` check by having the an empty handler installed? Same for the other two `is callable` checks.

Reply:
Yes I will add a card for refactoring of this.